pkg/sql/opt/xform/testdata/rules/limit

```diff
@@ -1364,6 +1364,72 @@ limit
  │              └── fd: ()-->(11), (12)-->(13,14)
  └── 5
 
+# Span boundary is exclusive, but the rule applies since the key is longer than
+# the prefix.
+opt expect=SplitScanIntoUnionScans
+SELECT id, latitude, longitude FROM index_tab
+WHERE (latitude = 0 OR latitude = 10) AND longitude IS NOT NULL
+ORDER BY longitude LIMIT 10
+----
+limit
+ ├── columns: id:1!null latitude:4!null longitude:5!null
+ ├── internal-ordering: +5
+ ├── cardinality: [0 - 10]
+ ├── key: (1)
+ ├── fd: (1)-->(4,5)
+ ├── ordering: +5
+ ├── union-all
+ │    ├── columns: id:1!null latitude:4!null longitude:5!null
+ │    ├── left columns: id:11 latitude:14 longitude:15
+ │    ├── right columns: id:21 latitude:24 longitude:25
+ │    ├── cardinality: [0 - 20]
+ │    ├── ordering: +5
```

Comment:
Looks like this test output is failing. A `sort` is added above the `union-all`. Perhaps because `release-21.1` is missing some of your recent set-related orderings commits?

Reply:
```
optimizer_test.go:231:
testdata/rules/limit:1369: SELECT id, latitude, longitude FROM index_tab
WHERE (latitude = 0 OR latitude = 10) AND longitude IS NOT NULL
ORDER BY longitude LIMIT 10
output didn't match expected:
@@ -3,30 +3,29 @@
├── internal-ordering: +5
├── cardinality: [0 - 10]
├── key: (1)
├── fd: (1)-->(4,5)
├── ordering: +5
- ├── union-all
+ ├── sort
│    ├── columns: id:1!null latitude:4!null longitude:5!null
- │    ├── left columns: id:11 latitude:14 longitude:15
- │    ├── right columns: id:21 latitude:24 longitude:25
│    ├── cardinality: [0 - 20]
│    ├── ordering: +5
│    ├── limit hint: 10.00
- │    ├── scan index_tab@d
- │    │    ├── columns: id:11!null latitude:14!null longitude:15!null
- │    │    ├── constraint: /14/15/16/17/11: (/0/NULL - /0]
- │    │    ├── limit: 10
- │    │    ├── key: (11)
- │    │    ├── fd: ()-->(14), (11)-->(15)
- │    │    ├── ordering: +15 opt(14) [actual: +15]
- │    │    └── limit hint: 10.00
- │    └── scan index_tab@d
- │         ├── columns: id:21!null latitude:24!null longitude:25!null
- │         ├── constraint: /24/25/26/27/21: (/10/NULL - /10]
- │         ├── limit: 10
- │         ├── key: (21)
- │         ├── fd: ()-->(24), (21)-->(25)
- │         ├── ordering: +25 opt(24) [actual: +25]
- │         └── limit hint: 10.00
+ │    └── union-all
+ │         ├── columns: id:1!null latitude:4!null longitude:5!null
+ │         ├── left columns: id:11 latitude:14 longitude:15
+ │         ├── right columns: id:21 latitude:24 longitude:25
+ │         ├── cardinality: [0 - 20]
+ │         ├── scan index_tab@d
+ │         │    ├── columns: id:11!null latitude:14!null longitude:15!null
+ │         │    ├── constraint: /14/15/16/17/11: (/0/NULL - /0]
+ │         │    ├── limit: 10
+ │         │    ├── key: (11)
+ │         │    └── fd: ()-->(14), (11)-->(15)
+ │         └── scan index_tab@d
+ │              ├── columns: id:21!null latitude:24!null longitude:25!null
+ │              ├── constraint: /24/25/26/27/21: (/10/NULL - /10]
+ │              ├── limit: 10
+ │              ├── key: (21)
+ │              └── fd: ()-->(24), (21)-->(25)
└── 10
--- FAIL: TestRules/limit (0.16s)
```